src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/config/WebClientConfiguration.kt

```diff
@@ -14,21 +15,60 @@ import org.springframework.web.reactive.function.client.WebClient
 
 @Configuration
 class WebClientConfiguration(
+  private val webClientBuilder: WebClient.Builder,
   @Value("\${oauth.endpoint.url}") private val oauthRootUri: String,
+  @Value("\${prison.api.url}") private val prisonApiBaseUrl: String,
   @Value("\${basm.endpoint.url}") private val basmRootUri: String,
 ) {
 
+  @Bean
+  fun oauthApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("hmpps-auth-api")
+
+    val exchangeStrategies = ExchangeStrategies.builder()
+      .codecs { configurer: ClientCodecConfigurer -> configurer.defaultCodecs().maxInMemorySize(-1) }
+      .build()
+
+    return webClientBuilder
+      .baseUrl(oauthRootUri)
+      .apply(oauth2Client.oauth2Configuration())
+      .exchangeStrategies(exchangeStrategies)
+      .build()
+  }
+
   @Bean
   fun oauthApiHealthWebClient(): WebClient {
-    return WebClient.builder().baseUrl(oauthRootUri).build()
+    return webClientBuilder.baseUrl(oauthRootUri).build()
+  }
+
+  @Bean
+  fun prisonApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("prison-api")
+
+    val exchangeStrategies = ExchangeStrategies.builder()
+      .codecs { configurer: ClientCodecConfigurer -> configurer.defaultCodecs().maxInMemorySize(-1) }
```

Comment:
This is generally handled with a property these days: https://github.com/ministryofjustice/hmpps-template-kotlin/blob/main/src/main/resources/application.yml#L9
Unfortunately there is no equivalent to -1 = no max - at least there wasn't last time I checked.